components/salsa-2022/src/setter.rs

```diff
@@ -0,0 +1,39 @@
+use crate::input_field::InputFieldIngredient;
+use crate::{durability, AsId, Durability, Runtime};
+use std::hash::Hash;
+
+#[must_use]
+pub struct Setter<'setter, K, F> {
+    runtime: &'setter mut Runtime,
+    key: K,
+    ingredient: &'setter mut InputFieldIngredient<K, F>,
+    durability: Durability,
+}
+
+impl<'setter, K, F> Setter<'setter, K, F>
+where
+    K: Eq + Hash + AsId,
+{
+    pub fn new(
+        runtime: &'setter mut Runtime,
+        key: K,
+        ingredient: &'setter mut InputFieldIngredient<K, F>,
+    ) -> Self {
+        Setter {
+            runtime,
+            key,
+            ingredient,
+            durability: Durability::LOW,
+        }
+    }
+
+    pub fn with_durability(self, durability: Durability) -> Self {
```

Comment:
Nit: we don't have any tests of durability! Maybe you want to add one? That said, I'm ok to land this PR and port the older tests over as follow-up.

Reply:
That's #339 so I'd like to leave it to a follow-up PR. I'll see if i can get it right.